src/mappings/utils/cumulativeVolumes.ts

```diff
@@ -201,3 +205,131 @@ export async function updateCumulativeVolumesForCurrencyPair(
+            // done: exit loop
+            break;
+        }
+
+        let amountToAdd: bigint | undefined = undefined;
+        if (pooledAmount.token === swapDetails.from.currency) {
+            amountToAdd = swapDetails.from.atomicAmount;
+            foundFrom = true;
+        } else if (pooledAmount.token === swapDetails.to.currency) {
+            amountToAdd = swapDetails.to.atomicAmount;
+            foundTo = true;
+        }
+
+        // update volume in place (ie. modify the entity directly)
+        if (amountToAdd) {
+            const newAmount = pooledAmount.amount + swapDetails.to.atomicAmount;
+            pooledAmount.amount = newAmount;
+            pooledAmount.amountHuman = await convertAmountToHuman(pooledAmount.token, newAmount);
+        }
+    }
+
+    // if we get here, there is at least one new pooled amount to add to the entity
+    if (!foundFrom) {
+        const pooledAmount = await createPooledAmount(swapDetails.from);
+        entity.amounts.push(pooledAmount);
+    }
+    if (!foundTo) {
+        const pooledAmount = await createPooledAmount(swapDetails.to);
+        entity.amounts.push(pooledAmount);
+    }
+
+    return entity;
+}
+
+async function fetchOrCreateEntity(
+    entityId: string, 
+    poolId: string, 
+    poolType: PoolType, 
+    tillTimestamp: Date, 
+    store: Store, 
+    entityBuffer: EntityBuffer
+): Promise<CumulativeDexTradingVolumePerPool> {
+    // fetch from buffer if it exists
+    return (entityBuffer.getBufferedEntityBy(
+            CumulativeDexTradingVolumePerPool.name,
+            entityId
+        ) as CumulativeDexTradingVolumePerPool | undefined) ||
+        // if not found, try to fetch from store
+        (await store.get(CumulativeDexTradingVolumePerPool, entityId)) ||
+        // still not found, create a new entity
+        new CumulativeDexTradingVolumePerPool({
+            id: entityId,
+            poolId: poolId,
+            poolType,
+            tillTimestamp,
+            amounts: []
+        });
+}
+
+export async function updateCumulativeDexVolumesForStandardPool(
+    store: Store,
+    timestamp: Date,
+    swapDetails: SwapDetails,
+    entityBuffer: EntityBuffer
+): Promise<CumulativeDexTradingVolumePerPool> {
+    const poolType = PoolType.Standard;
+
+    const poolId = inferGeneralPoolId(swapDetails.from.currency, swapDetails.to.currency);
+
+    const entityId = `${poolId}-${poolType}-${timestamp
+        .getTime()
+        .toString()}`;
+
+    const entity = await fetchOrCreateEntity(entityId, poolId, poolType, timestamp, store, entityBuffer);
+    return await updateOrAddPooledAmounts(entity, swapDetails);
+}
+
+export async function updateCumulativeDexVolumesForStablePool(
+    store: Store,
+    timestamp: Date,
+    poolId: number,
+    swapDetails: SwapDetails,
+    entityBuffer: EntityBuffer
+): Promise<CumulativeDexTradingVolumePerPool> {
+    const poolType = PoolType.Stable;
+
+    const poolIdString = `poolId_${poolId}`;
```

Comment:
Similar to previous comment, there is `inferGeneralPoolId` function - same could be added for stable pools for the sake of maintainability.

Reply:
> Similar to previous comment, there is `inferGeneralPoolId` function - same could be added for stable pools for the sake of maintainability.

Stable pools already come with an ID value so squid can use that, but general pools don't so it needs to infer a unique identifier.